src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -701,6 +701,8 @@ def _set_names(self, values: list) -> None:
         ------
         TypeError if each name is not hashable.
         """
+        if not is_list_like(values):
```

Comment:
can we have a test like index.names = pd.Index(...)? I try to make sure `is_list_like` won't trigger evaluation.

Reply:
Eager evaluation is triggered - 4 queries are run.
1 query to find the length of names
2 queries to create the required internal frame, one of them is for `to_pandas`
1 query to perform evaluation of result (another `to_pandas` query)

Reply:
Can we save one query in `_set_names` to convert the index to list first and then call set_index_names?